co-authors-plus.php

```diff
@@ -1050,24 +1050,112 @@ function filter_wp_get_object_terms( $terms, $object_ids, $taxonomies, $args ) {
 	function filter_count_user_posts( $count, $user_id ) {
 		$user = get_userdata( $user_id );
 
-		if ( is_object( $user ) ) {
+		// Return combined post count, if account is linked.
+		if ( $this->has_linked_account( $user ) ) {
+			return $this->get_combined_post_count( $this->get_coauthor_by( 'user_nicename', $user->user_nicename ) );
+		}
+
+		// Return term count, if account is not linked and term count is available.
+		if ( $this->has_author_terms( $user ) ) {
 			$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
 
-			$term = $this->get_author_term( $user );
+			if ( ! is_object( $user ) ) {
+				return $count;
+			}
 
-			if ( $term && ! is_wp_error( $term ) ) {
-				if ( 'guest-author' === $user->type ) {
-					// If using guest author term count, add on linked user count.
-					$count = (int) $count + $term->count;
-				} else {
-					$count = $term->count;
-				}
+			if ( $this->get_author_term( $user ) ) {
+				return $term->count;
 			}
 		}
 
+		// Return $count as fallback.
 		return $count;
 	}
 
+	/**
+	 * Check if user is linked to another account.
+	 *
+	 * @param WP_User $user The WP_User object.
+	 *
+	 * @return bool True on success, false on failure.
+	 */
+	function has_linked_account( $user ) {
+		$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
+		return (bool) $user->linked_account;
+	}
+
+	/**
+	 * Get the combined post count of the user and its linked coauthor account.
+	 *
+	 * @param WP_User $user The WP_User object.
+	 *
+	 * @return int The combined post count.
+	 */
+	function get_combined_post_count( $user ) {
+		global $wpdb;
+
+		$query = $wpdb->prepare(
+			"
+			SELECT
+				COUNT( {$wpdb->prefix}posts.ID ) as post_count
+			FROM
+				{$wpdb->prefix}posts
+			INNER JOIN
+				{$wpdb->prefix}term_relationships
+			ON
+				{$wpdb->prefix}posts.ID = {$wpdb->prefix}term_relationships.object_id
+			INNER JOIN
+				{$wpdb->prefix}terms
+			ON
+				{$wpdb->prefix}term_relationships.term_taxonomy_id = {$wpdb->prefix}terms.term_id
+			INNER JOIN
+				{$wpdb->prefix}users
+			ON
+				{$wpdb->prefix}posts.post_author = {$wpdb->prefix}users.ID
+			WHERE
+				{$wpdb->prefix}posts.post_type = 'post'
+			AND
+				(
+					{$wpdb->prefix}users.user_login = %s
+				OR
+					{$wpdb->prefix}terms.name = %s
+				)
+			GROUP BY
+				{$wpdb->prefix}posts.ID
+			",
+			$user->linked_account,
+			$user->user_nicename
+		);
+
+		return count( $wpdb->get_results( $query ) );
```

Comment:
Is there a way we could avoid direct DB queries? Something like this would negatively affect performance.

Reply:
I created that additional SQL query for a slight different use case. At the moment, there are three different cases in which the post count is incorrect:

1. A user publishes a post on behalf of a guest author whose account is not linked to a user account. In this case, the post count of the editor will be increased as well, even though the user is not co-author. For this case, I created https://github.com/Automattic/Co-Authors-Plus/issues/780.
2. A user and a guest author, that is not linked to another user account, create a post together. Both see a post count of 1. If the guest account gets linked to the user account, both accounts show a post count of 2 then. For this case, I created https://github.com/Automattic/Co-Authors-Plus/issues/779.
3. A user creates two posts and a guest author creates one post. If the guest account gets linked to the user account, both accounts should show a post count of 3. However, only the user account shows a post count of 3 while the guest account shows a post count of 5. This PR takes care of this third case.

I hope that it's a bit more clear while I went the sidetrack with the additional SQL query. That said, I shouldn't have done that and should have created separate issues for the other two problem cases initially.